### DISCUSSION TITLE
Feature Request:  Further customise of our .lintr file

### DISCUSSION THREAD

**Discussion by manciniedoardo** [pharmaverse/admiral#2477] [Ideas]
### Feature Idea

What if we leveraged the functionality in `{lintr}` to make custom linters?

One example I can think of is related to our recent error messaging update. Now that we use `{cli}`, do we ever need to use `stop()` or `warning()`? If not, we could make a custom linter for them to avoid ever introducing them in our codebase.

@pharmaverse/admiral any thoughts?

Some thoughts:
- This is of course relevant for `{admiraldev}` as well. 
- If we go ahead with this, it would be good to include some documentation about this. And also to include it maybe in the ext packages to try and enforce a certain way of doing things?

**Comment by bms63** [comment]
I'm curious how we can make this work for us especially since we are reaching more of a steady state.  This way some of our programming conventions can be automatically checked and enforced.

.... but I feel like it should be a discussion rather than an issue.

**Reply by bundfussr** [reply]
This could be particularly useful for extension packages.

- We would need to find a way to centralize the customizations. At the moment we have a `.lintr` file in each package. If we copy the customizations to each package, it will be hard to maintain.
- If one of the custom linters fails, it should be clear from the message that it is an admiral specific one. Otherwise it could be confusing if using `warning()` is marked as an issue.

**Reply by manciniedoardo** [reply]
I guess we could control it through the `.lintr` file in the `{admiraltemplate}` repo?

**Reply by bms63** [reply]
We could have a central `.lintr` file in `admiraldev` that the workflow grabs when doing the check.

**Reply by bundfussr** [reply]
> We could have a central `.lintr` file in `admiraldev` that the workflow grabs when doing the check.

If this is possible, I would prefer it. It seems better maintainable to me than a `.lintr` file in `{admiraltemplate}`.

**Reply by bms63** [reply]
@cicdguy possible to maintain a central `.lintr` file in `admiraldev` or in `admiralci` that all admiral repos use?

**Comment by bms63** [comment]
@ddsjoberg do the R cool kids do anything special with lintrs?   Does your packages use lintr at all?

We were looking at customizing ours a bit more to enforce some coding standards - applied to all the packages.

**Reply by ddsjoberg** [reply]
Hello @bms63 ! They are used in many many projects.

I don't use the lintrs myself, and rely on styler more. I do this because it's less heavy-handed. If you do implement, I would recommend to start light. There are few things more annoying than going to contribute to a project, and spend 10x more time dealing with the linting rules than the actual update you've made.

**Reply by bms63** [reply]
Thanks @ddsjoberg!!  

I think the checks would be very minimal.

**Comment by manciniedoardo** [comment]
Update: as @ddsjoberg suggested, I've started light. See here https://github.com/pharmaverse/admiraldev/pull/507. @pharmaverse/admiral , do you have any further ideas on what else we could add, also to promote standardisation across our core and ext packages? Otherwise we can go ahead and review/merge and start implementing across admiral and ext pkgs.

**Reply by bms63** [reply]
good topic for monday's meeting.  also nice to show how it works, how to add, etc

**Reply by manciniedoardo** [reply]
Yes, have already added it!

**Comment by manciniedoardo** [comment]
Closing as this is now implemented (#2863, #2913)
